x-pack/plugins/ml/server/models/model_management/models_provider.ts

```diff
@@ -516,4 +526,38 @@ export class ModelsProvider {
 
     return requestedModel || recommendedModel || defaultModel!;
   }
+
+  /**
+   * Puts the requested ELSER model into elasticsearch, triggering elasticsearch to download the model.
+   * Assigns the model to the * space.
+   * @param modelId
+   * @param mlSavedObjectService
+   */
+  async installElasticModel(modelId: string, mlSavedObjectService: MLSavedObjectService) {
+    const availableModels = await this.getModelDownloads();
+    const model = availableModels.find((m) => m.name === modelId);
+    if (!model) {
+      throw Boom.notFound('Model not found');
+    }
+
+    let esModelExists = false;
+    try {
+      await this._client.asInternalUser.ml.getTrainedModels({ model_id: modelId });
+      esModelExists = true;
+    } catch (error) {
+      // model doesn't exist, ignore error
+    }
```

Comment:
should we check the error code just in case? otherwise, we can silently skip a relevant error

Reply:
Updated in https://github.com/elastic/kibana/pull/169939/commits/c0fb0e92827f61b8b363ebe3f5e7dc634e73933a